File: sorting/sorting.java
import java.util.Arrays;

public class sorting {
    public static void main(String[] args) {
        int max_size = 10;
        int list[] = new int[max_size];
        
        Sort s = new Sort();

        for(int i=0; i<max_size;i++ ) list[i] = i;
        System.out.println("Before Selection : " + Arrays.toString(list));
        s.Selection(list);
        System.out.println(" After Selection : " + Arrays.toString(list));

        for(int i=0; i<max_size;i++ ) list[i] = i;
        System.out.println("Before Insertion : " + Arrays.toString(list));
        s.Insertion(list);
        System.out.println(" After Insertion : " + Arrays.toString(list));

        for(int i=0; i<max_size;i++ ) list[i] = i;
        System.out.println("Before Bubble : " + Arrays.toString(list));
        s.Bubble(list);
        System.out.println(" After Bubble : " + Arrays.toString(list));
    }
}

class Sort {
    public void Selection(int[] arr) {
        int N = arr.length;
        for(int i = 0 ; i < N ; i++) {
            int min = i;
            for(int j = i+1 ; j < N ;j++ ) {
                if(less(arr[j], arr[min])) {
                    min = j;
                }
            }
            exch(arr, i, min);
        }
    }
    public void Insertion(int[] arr) {

    }
    public void Bubble(int[] arr) {

    }
    private boolean less(Comparable v, Comparable w) {
        return v.compareTo(w) < 0;
    }
    private void exch(int[] arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }
}

File: sorting.c
#include <stdio.h>
#define MAX_SIZE 10

void selection(int arr[], int n) {
    int i, j, temp;
    for(i = 0;i<n;i++) {
        int min = i;
        for(j = i+1 ; j < n ;j++) {
            if(arr[j] < arr[min]) {
                min = j;
            }
        }
        temp = arr[min];
        arr[min] = arr[i];
        arr[i] = temp;
    }
}

void insertion(int arr[], int n) {
    int i, j, temp;
    for(i =0;i<n;i++) {
        for(j = i;j>0;j--){
            if(arr[j-1] > arr[j]) {
                temp = arr[j-1];
                arr[j-1] = arr[j];
                arr[j] = temp;
            }
            else {
                break;
            }
        }
    }
}

void bubble(int arr[], int n) {
    int i, j, temp;
    for(i = 0; i <n;i++) {
        for(j = 1;j<n-i;j++){
            if(arr[j-1] > arr[j]) {
                temp = arr[j-1];
                arr[j-1] = arr[j];
                arr[j] = temp;
            }
        }
    }
}

int main() {
    int n = MAX_SIZE;
    int list[MAX_SIZE];
    int i;

    for(i = 0 ; i < n ; i++ ) list[i] = n-i;
    printf("selection before : ");
    for(i = 0 ; i < n ; i++ ) printf("%d ", list[i]);
    printf("\n");
    selection(list, n);
    printf("selection after : ");
    for(i = 0 ; i < n ; i++ ) printf("%d ", list[i]);
    printf("\n");

    for(i = 0 ; i < n ; i++ ) list[i] = n-i;
    printf("insertion before : ");
    for(i = 0 ; i < n ; i++ ) printf("%d ", list[i]);
    printf("\n");
    insertion(list, n);
    printf("insertion after : ");
    for(i = 0 ; i < n ; i++ ) printf("%d ", list[i]);
    printf("\n");

    for(i = 0 ; i < n ; i++ ) list[i] = n-i;
    printf("bubble before : ");
    for(i = 0 ; i < n ; i++ ) printf("%d ", list[i]);
    printf("\n");
    bubble(list, n);
    printf("bubble after : ");
    for(i = 0 ; i < n ; i++ ) printf("%d ", list[i]);
    printf("\n");

    return 0;
}